// selenium-pages/dist/esm/index.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { Builder } from "selenium-webdriver";
import { Pages } from "./pages/";
export { Pages, WebElementEx, Dictionary } from "./pages/";
export class Selen {
    static addPath(path) {
        const sep = process.platform == "win32" ? ";" : ":";
        process.env.PATH += `${sep}${path}`;
    }
    static build(browser) {
        return __awaiter(this, void 0, void 0, function* () {
            return new Builder().forBrowser(browser).build();
        });
    }
}
Selen.Builder = Builder;
Selen.Pages = Pages;
