// selenium-pages/dist/cjs/index.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Selen = exports.Dictionary = exports.WebElementEx = exports.Pages = void 0;
const selenium_webdriver_1 = require("selenium-webdriver");
const pages_1 = require("./pages/");
var pages_2 = require("./pages/");
Object.defineProperty(exports, "Pages", { enumerable: true, get: function () { return pages_2.Pages; } });
Object.defineProperty(exports, "WebElementEx", { enumerable: true, get: function () { return pages_2.WebElementEx; } });
Object.defineProperty(exports, "Dictionary", { enumerable: true, get: function () { return pages_2.Dictionary; } });
class Selen {
    static addPath(path) {
        const sep = process.platform == "win32" ? ";" : ":";
        process.env.PATH += `${sep}${path}`;
    }
    static build(browser) {
        return __awaiter(this, void 0, void 0, function* () {
            return new selenium_webdriver_1.Builder().forBrowser(browser).build();
        });
    }
}
exports.Selen = Selen;
Selen.Builder = selenium_webdriver_1.Builder;
Selen.Pages = pages_1.Pages;
